### PR TITLE
Add 'now' flag to jobs

### LINKED ISSUE #109: Add 'now' flag to job entries
Proposed by @ryansanford as a preemptive production support feature.

Add a boolean `now` flag to job objects.
Toggling this flag will cause the job to run as soon as possible.

Change the job modification (post/put) route to only allow superusers to manipulate the field.

Change the `next` route to make two queries - first, check for any jobs with a `now` flag, second, the normal query that exists today. Critically, the `now` query should sort by the `modified` timestamp in the *opposite* order the normal query uses - that is, return a job that has **most** recently been modified and has a `now` flag. This contrasts with the normal query which is FIFO.

Should be set on endpoint tracked by #319.

### PR DESCRIPTION
**Pull request by nagem**
Closes #109

### CODE REVIEW

File: api/jobs/jobs.py [hunk 1]
```diff
@@ -46,14 +46,14 @@ def __init__(self, name, inputs, destination=None, tags=None, attempt=1, previou
 
         # TODO: validate inputs against the manifest
 
-        now = datetime.datetime.utcnow()
+        time_now = datetime.datetime.utcnow()
```
Comment (kofalt) [inline api/jobs/jobs.py]: Why was this renamed?

File: api/jobs/jobs.py [hunk 2]
```diff
@@ -78,6 +78,7 @@ def __init__(self, name, inputs, destination=None, tags=None, attempt=1, previou
         self.request         = request
         self.id_             = id_
         self.config          = config_
+        self.now             = now
```
Comment (kofalt) [inline api/jobs/jobs.py]: This looks to be using the old name?

**Comment by kofalt** [comment]
Could we remove `now` from `map()` if `now` is False?

**Comment by kofalt** [comment]
Should we add an index to the `now` flag? Would that first query result in a table scan?

(I assume we already have an index on the `modified` flag but I did not check.)

**Comment by kofalt** [comment]
LGTM after comments. Did not test.

**Comment by nagem** [comment]
Added a compound index that will support the `api/jobs/next` endpoint. @ryansanford let me know if you want to do any of your own testing, otherwise ready to merge.

**Comment by ryansanford** [comment]
@nagem I'm testing now...

**Comment by ryansanford** [comment]
@nagem Jobs without the now flag set are not being returned by `jobs/next` .  Also, new jobs from the UI and resulting from the dicom2nii gear don't appear to have that flag set in the db.

**Comment by kofalt** [comment]
@ryansanford I wouldn't expect jobs queued by the UI and rules to have a now flag, since the intention is that it's only used for administrative overrides.

**Comment by nagem** [comment]
@ryansanford bug fixed, I had removed the flag if it was `False` due to @kofalt's request, but my jobs used for testing still had the flag.

**Comment by ryansanford** [comment]
LGTM
